ledger/cow.go

```diff
@@ -140,8 +140,12 @@ func (cb *roundCowState) put(addr basics.Address, old basics.AccountData, new ba
 	}
 }
 
-func (cb *roundCowState) addTx(txn transactions.Transaction) {
-	cb.mods.Txids[txn.ID()] = txn.LastValid
+func (cb *roundCowState) addTx(txn transactions.Transaction, cachedTxid transactions.Txid) {
+	if cachedTxid == (transactions.Txid{}) {
```

Comment:
nit: you could eliminate this conditional check if you were to add the `cachedTxid = txn.ID()` on the `if !eval.validate` condition in the BlockEvaluator::transaction.

Reply:
+1, no conditions => better chance for inlining

Reply:
good suggestion, done.